src/components/Registration.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { createRoot } from 'react-dom/client';
+import './style.css';
+import { NameFromEmail } from '../handlers/userHandlers';
+
+
+
+const Registration = (props) => {
+    const [user, setUser] = useState({
+        username: '',
+        email: '',
+        password: '',
+        passwordConfirm: '',
+      });
+
+      function saveUser(){
+          event.preventDefault();
+          if(user.email.indexOf('@') <= -1) throw console.log("email non valid");
```

Comment:
myslím že tu nechceme použít "throw". Throw vyhodí nějakou vyjímku, navíc v tomhle případě výsledek console.log což je undefined. Tj buĎ něco jako
```
if(user.email.indexOf('@') <= -1) throw new Error("email non valid");
```
ale musela bys to ještě někde odchytávat (try/catch) což se tady neděje (resp probublá to do onClick buttonu který s tím nic neudělá). Tj úplně by stačilo

```
if(user.email.indexOf('@') <= -1) {
   console.log("email non valid")
   return;
}
```